### DISCUSSION TITLE
Image Handling question

### DISCUSSION THREAD

**Discussion by IsaaacD** [PiranhaCMS/piranha.core#2073] [Q&A]
On the site Piranha CMS mentions IMAGE HANDLING: Built-in media library with scaling and cropping without losing focus of the important parts of you images.

I've included `options.UseImageSharp();` but I'm wondering is there somethign else I need to do? Do I need to configure something, my goal is for the user I'm creating this site for to be able to crop images in the Manager section so they don't need to be resized in another editor. Maybe this isn't possible or I need to develop something, but want to see what the options are.

Thanks

**Comment by tidyui** [comment]
The image functionality included is very "developer centric", in other words its intention is to make it easy for the editor to just upload and use any image and easy for you as the developer to make sure the image is cropped and scaled to the correct size for any given context. For example, given any image field you can supply a `width` to scale it, or a `width` and `height` to scale & crop the image. 

For the `ImageBlock` there are also some tools for the editor to specify the aspect ratio of the image that you can then use by calling the `IMediaHelper.Resize(...)` method. As an example you can see this in use in the display template for the `ImageBlock` in the template site here:

https://github.com/PiranhaCMS/piranha.core.templates/blob/master/src/web/mvc/Views/Cms/DisplayTemplates/ImageBlock.cshtml#L3

Documentation on the media helper can be found here:

https://piranhacms.org/docs/master/application/services/application-service

**Reply by IsaaacD** [reply]
Thanks for the info! If I were to create a module with a mini resizer in JS, would it be something I could issue a PR? I wouldn't want to maintain it necessarily and I think having this type of feature in manager would be amazing. Could set jpeg compression and make multiple version available to the users of Piranha
